### ISSUE TITLE
fc.yahoo.com not responding

### ISSUE THREAD

**Issue by helphp** [dpguthrie/yahooquery#241]
fc.yahoo.com not responding for me. It's used in the setup_session_with_cookies_and_crumb() function in utils/\_\_init\_\_.py before doing anything else. I temporarily changed line 176 to use the URL https://finance.yahoo.com/quote/TSLA/ instead and it seems to be working for now.

**Comment by RudyNL** [comment]
Same problems in the Netherlands. I had the same results as @jirisarri10 at Decemeber 5 0:30 CET.
A VPN to simulate the US location doesn't work. Finally made the change as proposed by @helphp 
```
    try:
        response = session.get('https://finance.yahoo.com/quote/TSLA/')
#        response = session.get('https://fc.yahoo.com')
    except Exception:

```
in `......../site-packages/yahooquery/utils/__init__.py`. My own program is running again but has missing data problems. The test of @jirisarri10 has the same outcome `{'ORCL': 'Invalid cookie'}`. Its my bedtime, tomorrow I will look at it again.

**Comment by 0zd3m1r** [comment]
![image](https://github.com/dpguthrie/yahooquery/assets/18439783/c305ac2f-a19c-4456-9e67-3f947995aa96)

yes, yahooquery is currently not working.

**Comment by RudyNL** [comment]
The Yahoo servers seem to work again and there are now problems again. My program did run correctly from December 5 6:15 till 8:15 CET in the Netherlands. And now around 8:45 CET I am having the same issues as yesterday. It looks like Yahoo is working on it.

**Comment by ibart** [comment]
> It looks like Yahoo is working on it.

They are working on it since october...

**Comment by RudyNL** [comment]
Is the patch really working? For me it isn't doing anything.
I am doing the @jirisarri10 test and with or without the patch the result is the same: `{'ORCL': 'Invalid cookie'}`
`ticker.history()`
Is working with and without the patch.

**Comment by newspapermann** [comment]
not working for me either

**Comment by nathanramoscfa** [comment]
I went to fc.yahoo.com in my browser and got this message:

![image](https://github.com/dpguthrie/yahooquery/assets/86393277/e2f29e4d-6dfd-47d0-b8e0-aed8217f3232)

I tried `yfinance` library to see if maybe that library was having the same problems but it worked with no problems:

```
import yfinance as yf
orcl = yf.Ticker("ORCL").info
orcl
```

```
{'address1': '2300 Oracle Way',
 'city': 'Austin',
 'state': 'TX',
 'zip': '78741',
 'country': 'United States',
 'phone': '737-867-1000',
 'website': 'https://www.oracle.com/',
 'industry': 'Software - Infrastructure',
 'industryKey': 'software-infrastructure',
 'industryDisp': 'Software - Infrastructure',
 'sector': 'Technology',
 'sectorKey': 'technology',
 'sectorDisp': 'Technology',
...
```

**Comment by chfiii** [comment]
The patch that HELPHP suggested to change line 177 in yahooquery/utils/__init__py from:
"https://fc.yahoo.com/" to "https://finance.yahoo.com/quote/TSLA" works fine for me.  (in CT, USA, yahooquery version 2.3.3)

**Comment by ibart** [comment]
> I went to fc.yahoo.com in my browser and got this message:
> 
> ![image](https://private-user-images.githubusercontent.com/86393277/288117674-e2f29e4d-6dfd-47d0-b8e0-aed8217f3232.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDE3OTYyMjksIm5iZiI6MTcwMTc5NTkyOSwicGF0aCI6Ii84NjM5MzI3Ny8yODgxMTc2NzQtZTJmMjllNGQtNmRmZC00N2QwLWI4ZTAtYWVkODIxN2YzMjMyLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzEyMDUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMjA1VDE3MDUyOVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTQ4YTNiMTRiZGJjMmZiNzY2MTExMjc0MDk2M2Y1YTU0YjExOWRjNmU3ZjYwNmQ3YjFlN2ZhYmU4YjM4ODlhZTkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.3lfvcS30kPRDQIak_rbKzx3hnjqx3oGr0Daq-GpyldI)
> 
> I tried `yfinance` library to see if maybe that library was having the same problems but it worked with no problems:
> 
> ```
> import yfinance as yf
> orcl = yf.Ticker("ORCL").info
> orcl
> ```
> 
> ```
> {'address1': '2300 Oracle Way',
>  'city': 'Austin',
>  'state': 'TX',
>  'zip': '78741',
>  'country': 'United States',
>  'phone': '737-867-1000',
>  'website': 'https://www.oracle.com/',
>  'industry': 'Software - Infrastructure',
>  'industryKey': 'software-infrastructure',
>  'industryDisp': 'Software - Infrastructure',
>  'sector': 'Technology',
>  'sectorKey': 'technology',
>  'sectorDisp': 'Technology',
> ...
> ```

it used to has... I think they solved with cookie & crumb. 

This nightmare shows us that we need to find a more stable source of data. I would like to have this, with a subscription, by Yahoo Finance. Why the hell Yahoo is not understanding this? -_-

**Comment by RudyNL** [comment]
I am testing at December 5 18:40 CET in the Netherlands. With and without patch it isn't working. But with and without the patch its now working with a VPN into the USA.

My conclusion is that the best thing you can do is to wait. Yahoo is continuously changing the circumstances. Just notice that early this morning I could run a program retrieving data of about 1000 equities within two hours without any problems.

**Comment by SC8198** [comment]
Can confirm stopped working Dec 4th around 12:30 pm ~ MST

**Comment by SC8198** [comment]
> The patch that HELPHP suggested to change line 177 in yahooquery/utils/**init**.py from: "https://fc.yahoo.com/" to "https://finance.yahoo.com/quote/TSLA" works fine for me. (in CT, USA, yahooquery version 2.3.3)
> 
> I'm using both the price and history modules.

May I ask how you patched this? did you clone the repo and install from a directory? Just trying to get my script working thanks.

**Comment by chfiii** [comment]
The 'approved' way would be to clone the repo. I hacked quickly by doing a "pip show yahooquery" to get the directory and then just replaced the line.

As for the question about won't is just get Tesla data - no.  This call is just to establish the cookie and crumb with yahoo so the session can be setup.  I initially thought the same ;)

**Comment by itzmk21** [comment]
Can confirm, I get the same errors, 5th Dec, 08:11pm GMT(UTC+0), UK based.

**Comment by fredrik-corneliusson** [comment]
Using the suggested fix by changing URL on line 176 just makes it return "invalid cookie" in the data response.
`import yahooquery as yq
tq = yq.Ticker('TSLA')
print(tq.earnings)`
Output:
`{'TSLA': 'Invalid Cookie'}`

**Comment by quikksilver1** [comment]
At least you guys are getting responses back. I am calling the Ticker() class and then the history class and it just stalls out. I am doing this in a vacuum so I know my code is not interfering. Is it possible for them to block someone?

**Comment by ibart** [comment]
> At least you guys are getting responses back. I am calling the Ticker() class and then the history class and it just stalls out. I am doing this in a vacuum so I know my code is not interfering. Is it possible for them to block someone?

you are not banned. In case, you can just try a VPN. The response is the same for all of us: 

HTTPSConnectionPool(host='fc.yahoo.com', port=443): Max retries exceeded with url: / (Caused by ReadTimeoutError("HTTPSConnectionPool(host='fc.yahoo.com', port=443): Read timed out. (read timeout=5)"))

Even changing the timeout, the problem persists. I think we are having the cookie and crumbs problem yfinance used to have 1 month ago.

**Comment by no0be** [comment]
> Using the suggested fix by changing URL on line 176 just makes it return "invalid cookie" in the data response.
> 
> ```
> import yahooquery as yq
> tq = yq.Ticker('TSLA')
> print(tq.earnings)
> ```
> 
> Output: `{'TSLA': 'Invalid Cookie'}`

I confirm I observe the same behavior.

```
import json
from yahooquery import Ticker
tickers = Ticker(["TSLA"])
print(json.dumps(tickers.price,indent=3))
```

Output:
```
{
   "TSLA": "Invalid Cookie"
}
```

**Comment by nathanramoscfa** [comment]
I forked yahooquery and made the suggest change to `utils/__init__.py`:

![image](https://github.com/dpguthrie/yahooquery/assets/86393277/8df6cf34-eed1-4c4a-81db-dd641c35083f)

I can confirm that it is now working as of this morning.

**Comment by no0be** [comment]
> I forked yahooquery and made the suggested change to `utils/__init__.py`:
> 
> ![image](https://private-user-images.githubusercontent.com/86393277/288466367-8df6cf34-eed1-4c4a-81db-dd641c35083f.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDE4ODA0OTUsIm5iZiI6MTcwMTg4MDE5NSwicGF0aCI6Ii84NjM5MzI3Ny8yODg0NjYzNjctOGRmNmNmMzQtZWVkMS00YzRhLTgxZGItZGQ2NDFjMzUwODNmLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzEyMDYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMjA2VDE2Mjk1NVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTdkNjhiNWNhMTc1MWZlYjU0MzVjMmNjYzQwNjRmYWU4ZjhhMzkxNjhlMWJjNDllN2JjZGE0MDI1YjY5NjgzYmMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.ZidYtsEc0LpRipjAGPfoNxAEyYTsdxSbVGYb7uPF_s8)
> 
> I can confirm that it is now working as of this morning.

Do you need to do something to rebuild the package or something similar after the change? I'm running inside a venv build with pipenv and just changed the file as you did in `.venv/lib/python3.9/site-packages/yahooquery/utils/__init__.py` but it still does not work.

**Comment by fredrik-corneliusson** [comment]
Could you provide a example of the code that was solved by the suggested solution?
I suspect this "solution" just hides and/or solves a subset of the API-calls in yahooquery.
Is there integration test suite for yahoo query that tests against the live yahoo API we can use to verify what is working and what is broken?

**Comment by fredrik-corneliusson** [comment]
> @nathanramoscfa , could you provide a example of the code that was solved by the suggested solution? I suspect this "solution" just hides and/or solves a subset of the API-calls in yahooquery. Is there integration test suite for yahoo query that tests against the live yahoo API we can use to verify what is working and what is broken?

I just did a quick check, the fix "solves" that the ticker.history() hangs for a while before returning data. Probably due to waiting for fc.yahoo.com to time out. 
But ticker.history() returns data with and without the "solution", just much slower.
Earnings (and I suspect many other request) still returns invalid_cookie. 
This is probably due to the `response.cookies` is empty (RequestsCookieJar[]) for the suggested URL "solution":
`response = session.get('https://finance.yahoo.com/quote/TSLA/')`.

Test code:
```
import yahooquery as yq
tq = yq.Ticker('TSLA')
print(tq.history())
print(tq.earnings)
```

**Comment by RudyNL** [comment]
@fredrik-corneliusson I am using a private program analyzing about 1000 equities based on `yahooquery`. You can interpret this program as a stress test. It has been running this morning and yesterday evening after closure of Wall-Street using a VPN (Proton VPN on Linux) to simulate my location in the USA. I don't need the patch and there are no problems. The only difference with normally running is that the data retrieval is about a factor two slower. From the Netherlands without VPN only `history()` is working. All other methods are leading to error messages.

My conclusion is that the results are very location dependent. The current situation is for me stable for about one and a half day. Before that, the results were also very time dependent.

**Comment by fredrik-corneliusson** [comment]
Thanks for the info, see my comment I made just before yours for some more details (we posted almost at the same time). 
> My conclusion is that the results are very location dependent.
This seems to be very similar to the previous issue we had a couple of months ago. My guess is that yahoo rolls out changes to their infrastructure gradually in different regions. If that is correct it is probably just a matter of time before the VPN solution breaks as well.

**Comment by sixxfour** [comment]
> fc.yahoo.com not responding for me. It's used in the setup_session_with_cookies_and_crumb() function in utils/**init**.py before doing anything else. I temporarily changed line 176 to use the URL https://finance.yahoo.com/quote/TSLA/ instead and it seems to be working for now.

I ran into the same issue today and using 'finance.yahoo.com' worked for me as well. Thanks for posting!

**Comment by pkonduri** [comment]
Does anyone know of a long term fix here? Or will yahoo just make this an untenable option?

**Comment by dpguthrie** [comment]
My guess here is that at some point another issue will be introduced from yahoo's side.  It will continue to be like playing whack-a-mole.

With that said, I did just release 2.3.4.  Hopefully that helps some of you (tested in a few different areas and was able to get results back from each endpoint...so 🤷 )

**Comment by RudyNL** [comment]
Same here with version 2.3.5. In the Netherlands I still need a VPN to the USA.

**Comment by fredrik-corneliusson** [comment]
Thanks Doug for your relentless and splendid work!
2.3.5 got it working again from Amazon EC2 instance in US-West region, but not from EU-Stockholm region. So probably region sensitive issue.

**Comment by dpguthrie** [comment]
For those of you not in the U.S., when you navigate to yahoo finance, what is the site that you end up going to?  Is it "https://es.finance.yahoo.com", "https://de.finance.yahoo.com", "https://uk.finance.yahoo.com"...something else?  I'm wondering if you pass in a country from [here](https://github.com/dpguthrie/yahooquery/blob/master/yahooquery/utils/countries.py) to one where you're located or close by when initializing the `Ticker` if that helps at all.

**Comment by RudyNL** [comment]
I am having the same problems with version 2.3.5 with VPN into the USA. There is a lot of missing data. In my program all equities are checked for missing or wrong data data and in a second and third pass I am trying to retrieve the data again. This normally helps against random problems, but now the missing data is structural.

Version 2.3.5 in Europe doesn't work
```
>>> import yahooquery
>>> ticker=yahooquery.Ticker('ORCL')
>>> ticker.price
{'ORCL': 'Invalid Cookie'}

```
For production I am using version 2.3.3 with VPN into the USA and the patch
`        response = session.get('https://finance.yahoo.com')`
instead of
`        response = session.get('https://fc.yahoo.com')`
The extension `/quote/TSLA/` is superfluous. This is running slow (about a factor two) but fine.

Using the url `https://de.finance.yahoo.com` instead of `https://finance.yahoo.com` in the web-browser doesn't seem to make much difference. Only the news feed is in German and partly the information presented is translated into German. In my program I am cycling the location `country='spain'` hoping to alternate the used Yahoo servers. I never had any benefit of this process.

**Comment by dpguthrie** [comment]
Thanks for testing the country parameter.  I was trying to use that as a way to set the host name when obtaining cookies/crumb.  Doesn't appear to do anything, so I'll probably remove that piece of code then.

**Comment by kensthilaire** [comment]
@dpguthrie, I have updated to 2.3.5 and verified that all is well again for my application. I am in the eastern US. Thank you once again for the quick turn on the refactoring to whack this latest mole. 

And, thank you for adjusting the poetry file to allow python 3.8. Saved me from having my own version that relaxed the python version.

Ken

**Comment by SC8198** [comment]
> The 'approved' way would be to clone the repo. I hacked quickly by doing a "pip show yahooquery" to get the directory and then just replaced the line.
> 
> As for the question about won't is just get Tesla data - no. This call is just to establish the cookie and crumb with yahoo so the session can be setup. I initially thought the same ;)

THANK YOU!!!

**Comment by RudyNL** [comment]
There are two related issues at `yfinance` 
1. [ReadTimeoutError on fc.yahoo.com](https://github.com/ranaroussi/yfinance/issues/1765)
2. [Connection refused](https://github.com/ranaroussi/yfinance/issues/1750)

**Comment by ibart** [comment]
> For those of you not in the U.S., when you navigate to yahoo finance, what is the site that you end up going to? Is it "https://es.finance.yahoo.com", "https://de.finance.yahoo.com", "[https://uk.finance.yahoo.com"...something](https://uk.finance.yahoo.com%22...something) else? I'm wondering if you pass in a country from [here](https://github.com/dpguthrie/yahooquery/blob/master/yahooquery/utils/countries.py) to one where you're located or close by when initializing the `Ticker` if that helps at all.
> 
> ```python
> import yahooquery as yq
> 
> t = Ticker('aapl', country='spain')
> t.price
> ```
yq.Ticker ...

**Comment by ibart** [comment]
```
import yahooquery as yq

t = yq.Ticker('aapl')
t.price 
```
2.3.6, from Italy it fails: _Failed to obtain crumb.  Ability to retrieve data will be significantly limited._ BUT with a VPN from U.S.A. is working.

**Comment by RudyNL** [comment]
@dpguthrie Blocking the EU is quite popular. They are afraid of fines in case of breaking EU privacy and data collection rules. Recently [CNN](https://edition.cnn.com) blocked Europe.

Is it possible to find a solution with a Proxy server. I have been experimenting with it, but I am not experienced enough. A Proxy server only for `yahooquery` is to be preferred. With a VPN I must not open my mail because the mail providers starts to complain that I am at the wrong continent.

**Comment by dpguthrie** [comment]
@RudyNL Curious, are you able to use yfinance without vpn/proxy?  Or do you run into the same problem?

**Comment by dpguthrie** [comment]
Interesting that 2.3.5 is different from a data perspective.  I released 2.3.6 today and updated the session setup to just use finance.yahoo.com instead of the tsla endpoint.

**Comment by dpguthrie** [comment]
Cool.  Thanks for checking so quickly.

**Comment by james-stevens** [comment]
> Is it possible to find a solution with a Proxy server. I have been experimenting with it, but I am not experienced enough. A Proxy server only for `yahooquery` is to be preferred. With a VPN I must not open my mail because the mail providers starts to complain that I am at the wrong continent

https://www.scrapingbee.com/blog/python-requests-proxy/

You should be able to set two environment variables (before running your python code) to get python/requests to use an HTTP(S) proxy - that way only that program would use the proxy.

from that page, this example
```
$ export HTTP_PROXY='http://10.10.10.10:8000'
$ export HTTPS_PROXY='http://10.10.10.10:1212'
$ ./your_code.py
```
Obv changing the IP Address / Port numbers to point to your http(s) proxy

**Comment by RudyNL** [comment]
@dpguthrie I am experimenting a bit with version 2.3.6. The problems are caused by the method 'get_crumb'. Just replacing the exception handling by 'pass' gives this error
```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/rudy/.local/lib/python3.10/site-packages/yahooquery/ticker.py", line 93, in __init__
    super(Ticker, self).__init__(**kwargs)
  File "/home/rudy/.local/lib/python3.10/site-packages/yahooquery/base.py", line 946, in __init__
    self.crumb = get_crumb(self.session)
  File "/home/rudy/.local/lib/python3.10/site-packages/yahooquery/utils/__init__.py", line 1392, in get_crumb
    response = session.get("https://query2.finance.yahoo.com/v1/test/getcrumb")
  File "/home/rudy/.local/lib/python3.10/site-packages/requests/sessions.py", line 602, in get
    return self.request("GET", url, **kwargs)
  File "/home/rudy/.local/lib/python3.10/site-packages/requests/sessions.py", line 589, in request
    resp = self.send(prep, **send_kwargs)
  File "/home/rudy/.local/lib/python3.10/site-packages/requests/sessions.py", line 703, in send
    r = adapter.send(request, **kwargs)
  File "/home/rudy/.local/lib/python3.10/site-packages/yahooquery/utils/__init__.py", line 1341, in send
    return super(TimeoutHTTPAdapter, self).send(request, **kwargs)
  File "/home/rudy/.local/lib/python3.10/site-packages/requests/adapters.py", line 510, in send
    raise RetryError(e, request=request)
requests.exceptions.RetryError: HTTPSConnectionPool(host='query2.finance.yahoo.com', port=443): Max retries exceeded with url: /v1/test/getcrumb (Caused by ResponseError('too many 500 error responses'))
```
In a browser the url `https://query2.finance.yahoo.com/v1/test/getcrumb` is producing a crumb `szf9c5Bvejb`

**Comment by othalan** [comment]
I have had mixed results with the update....

(As a side note, thank you for the hard work keeping yahooquery functional!)

Version 2.3.6 ... A simple test Works from Thailand and over a VPN to the US, but does ***not*** work over a VPN to European countries (NordVPN)

The simple test:
```python
[nav] In [5]: import yahooquery as yq
         ...: t=yq.Ticker("AAPL", asynchronous=True)
```
The error I receive over a VPN to European countries (Spain, Romania and UK tested, via NordVPN):
```
RetryError: HTTPSConnectionPool(host='query2.finance.yahoo.com', port=443): Max retries exceeded with url: /v1/test/getcrumb (Caused by ResponseError('too many 500 error responses'))
```
For a more complex test, I get something ***very*** strange:

Version 2.3.6 ... My application has a segfault.
Version 2.3.5 ... My application works.

This is an old, large, complex, and badly written application. I have not yet been able to track down the source of the segfault, but I tend to doubt yahooquery is responsible looking at recent changes.

```
zsh: segmentation fault  ipython server.py
/Users/morrisde/.pyenv/versions/3.10.3/lib/python3.10/multiprocessing/resource_tracker.py:224: UserWarning: resource_tracker: There appear to be 1 leaked semaphore objects to clean up at shutdown
  warnings.warn('resource_tracker: There appear to be %d '
```

**Comment by RudyNL** [comment]
I am still searching for a solution. In my search I found:
https://stackoverflow.com/questions/76065035/yahoo-finance-v7-api-now-requiring-cookies-python
At the end is a piece of code which functions both in Europe and the USA
```
import requests

apiBase = 'https://query2.finance.yahoo.com'
headers = { 
  "User-Agent": 
  "Mozilla/5.0 (Windows NT 6.1; Win64; x64)"
}

def getCredentials(cookieUrl='https://fc.yahoo.com', crumbUrl=apiBase+'/v1/test/getcrumb'):
  cookie = requests.get(cookieUrl).cookies
  crumb = requests.get(url=crumbUrl, cookies=cookie, headers=headers).text
  return {'cookie': cookie, 'crumb': crumb}

def quote(symbols, credentials):
  url = apiBase + '/v7/finance/quote'
  params = {'symbols': ','.join(symbols), 'crumb': credentials['crumb']}
  response = requests.get(url, params=params, cookies=credentials['cookie'], headers=headers)
  quotes = response.json()['quoteResponse']['result']
  return quotes

credentials = getCredentials()
quotes = quote(['GOOG', 'TSLA'], credentials)
if quotes:
  for quote in quotes:
    print(f"{quote['symbol']} price is {quote['currency']} {quote['regularMarketPrice']}")

# GOOG price is USD 121.08
# TSLA price is USD 256.24
```
May be some elements can be transferred into `yahooquery`. I am wondering that `credentials['cookie'] `gives a cookie and `qoutes `give a bunch of equity information. Notice the prefix for a cookie is different from the prefix for a crumb.

I am continuing with experimenting.

**Comment by RudyNL** [comment]
I have placed the cookie and crumb management of the last code block of
https://stackoverflow.com/questions/76065035/yahoo-finance-v7-api-now-requiring-cookies-python
into the file `../utils/__init__.py` version v2.3.6 .
That means that the begin of the method `get_crumb` is now:
```
def get_crumb(session):
    cookieUrl= 'https://fc.yahoo.com'
    crumbUrl = 'https://query2.finance.yahoo.com/v1/test/getcrumb'
    try:
        session.cookies   = requests.get(cookieUrl).cookies
        response = session.get(crumbUrl,cookies=session.cookies)
    except (ConnectionError, RetryError):
```
Probably there is now a superfluous retrieval of the cookies. To solve that problem I need more understanding of the code.
Notice that the crumb url is `https://fc.yahoo.com`; `https://finance.yahoo.com` isn't working. Both is the USA (VPN) and Europe my personal programs are working again. The ticker retrieval is a bit slow.

**Comment by jgriessler** [comment]
I have the "crumb" issues as well (running yahooquery out of Germany) since yesterday. 
(newest yahooquery). Things worked fine until last week. 

I started a little bit of debugging in my browser. I think it's related to the "Consent" page that I get redirected to when first time going to finance.yahoo.com (after clearing the cookies & co in browser) Until you acknowledge that, the "https://query2.finance.yahoo.com/v1/test/getcrumb" doesn't return any crumb. 

So I tweaked the __init__ code to look for the redirection to the consent page and send the "accept" to get things working again. What's weird though is that while playing a little bit more today, at least the yahooquery stuff works again directly even without my additional adjustments. Firefox browser still get's me redirected to the "consent" page, but the yahooquery code doesn't get redirected to the "consent" page. 

For what it's worth - here the code tweak I added for the extra consent check (yfinance does something similar). 

`def setup_session(session: requests.Session):
    url = "https://finance.yahoo.com"
    try:
        response = session.get(url, allow_redirects=True)
    except SSLError:
        counter = 0
        while counter < 5:
            try:
                session.headers = random.choice(HEADERS)
                response = session.get(url, verify=False)
                break
            except SSLError:
                counter += 1

    if not isinstance(session, FuturesSession):

        # check for and handle consent page:w
        if response.url.find('consent'):
            logger.debug(f'Redirected to consent page: "{response.url}"')

            soup = BeautifulSoup(response.content, 'html.parser')
            
            params = {}
            for param in ['csrfToken', 'sessionId']:
                try:
                    params[param] = soup.find('input', attrs={'name': param})['value']
                except Exception as exc:
                    logger.critical(f'Failed to find or extract "{param}" from response. Exception={exc}')
                    return

            logger.debug(f'params: {params}')
               
            response = session.post(
                'https://consent.yahoo.com/v2/collectConsent',
                data={
                    'agree': ['agree', 'agree'],
                    'consentUUID': 'default',
                    'sessionId': params['sessionId'],
                    'csrfToken': params['csrfToken'],
                    'originalDoneUrl': url,
                    'namespace': 'yahoo'
                })
            # just assume things are fine and session is setup now

        return session

    _ = response.result()
    return session
`

**Comment by jgriessler** [comment]
Retested again, "crumb" error message is back when I use yahooquery to fetch historic quotes without the extra code above to handle the "Consent" page. 
So likely just a glitch with my testing. 
Things work fine with the extra code above to agree to the data protection consent page. Without I get the "crumb" error.
will watch it a little longer. Overall the fetching seems to be slower then I remember from the last few months. But that is not unexpected as each ticker I query is setting up a new session (incl. ack the consent)

**Comment by RudyNL** [comment]
Its now a fact that Yahoo didn't block Europe. If you are getting data depends on the data handling. So it must be possible to create a `yahooquery` which works both in Europe and the USA.

**Comment by jgriessler** [comment]
> Its now a fact that Yahoo didn't block Europe. If you are getting data depends on the data handling. So it must be possible to create a `yahooquery` which works both in Europe and the USA.

I opened a Feature request #247 to see what Doug thinks about this  (Assuming the crumb issue I see *now* is related to the Data-protection CONSENT page that Yahoo redirects European users).  PS:  I like yahooquery a lot - very robust and easy to use so far.

**Comment by james-stevens** [comment]
v2.3.7 just worked for me from the UK - yes, it takes a little longer, but I get all the prices I need in one call, so makes very little difference, TBH

I upgraded my server and my application is now back in business !!!

Thank you *SO* much !! :rocket:

**Comment by qkxie** [comment]
v2.3.7 always raise `Failed to obtain crumb.  Ability to retrieve data will be significantly limited.` and then `{'xxx.SZ': 'Invalid Cookie'}`. My region is Taiwan

**Comment by dexter31** [comment]
This might be an old issue, but resurfaced on the latest version for the last week(Stopped working as of 4th May). Switching VPN's to Singapore, India, Netherlands, US, UK, Malaysiya, HK and Australia did not work. 

To keep it simple, yahoo were always playing whack a mole and now have some method (eitheusing fingerprinting or something advanced that bot detection algos do) to figure a curl from a terminal and a brtqwoser. 

Simply put the https://query2.finance.yahoo.com/v1/test/getcrumb URL will give you a 429. 

Workaround - Im not going to make it easy for yahoo to figure how to protect this. But a fairly easy fix exists - but will be an addon to using yahooquery.
